File: app/src/main/java/se/gustavkarlsson/aurora_notifier/android/dagger/modules/SettingsModule.java
package se.gustavkarlsson.aurora_notifier.android.dagger.modules;

import android.content.Context;

import dagger.Module;
import dagger.Provides;
import dagger.Reusable;
import se.gustavkarlsson.aurora_notifier.android.settings.DebugSettings;
import se.gustavkarlsson.aurora_notifier.android.settings.Settings;

@Module
public abstract class SettingsModule {

	@Provides
	@Reusable
	static Settings provideSettings(Context context) {
		return new Settings(context);
	}

	@Provides
	@Reusable
	static DebugSettings provideDebugSettings(Context context) {
		return new DebugSettings(context);
	}

}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/modules/SettingsModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules

import android.content.Context
import android.content.SharedPreferences
import com.f2prateek.rx.preferences2.RxSharedPreferences
import dagger.Module
import dagger.Provides
import se.gustavkarlsson.skylight.android.services.Settings
import se.gustavkarlsson.skylight.android.services_impl.RxPreferencesSettings
import javax.inject.Singleton


@Module
class SettingsModule {

	@Provides
	@Singleton
	fun provideRxSharedPreferences(sharedPreferences: SharedPreferences): RxSharedPreferences =
		RxSharedPreferences.create(sharedPreferences)

	@Provides
	@Singleton
	fun provideSettings(context: Context, rxSharedPreferences: RxSharedPreferences): Settings {
		return RxPreferencesSettings(
			context,
			rxSharedPreferences
		)
	}
}
